src/tracking.ts

```diff
@@ -213,14 +213,18 @@ export class SearchIOAnalytics {
       if (lastEvent) {
         queryId = lastEvent.queryId;
       }
+      if (!queryId) {
+        console.error(
+          "No queryId found. Use updateQueryId to set the current queryId or call trackForQuery with a specific queryId."
+        );
+        return;
+      }
     }
 
     if (!queryId) {
-      console.error(
```

Comment:
You want to keep this `console.error` and wrap it in a condition to only error when it's `click`, `redirect`, or `promotion_click`:
```typescript
if (!queryId) {
  if (FUNNEL_ENTRY_TYPES.includes(type)) {
    console.error(
      "No queryId found. Use updateQueryId to set the current queryId or call trackForQuery with a specific queryId."
    );
  }
  return;
}
```